sdk/keyvault/keyvault-keys/src/localCryptography/providers.ts

```diff
@@ -0,0 +1,231 @@
+
+import { RSA_PKCS1_OAEP_PADDING, RSA_PKCS1_PADDING } from "constants";
+import { JsonWebKey, KeyOperation } from "../keysModels";
+import { createVerify, publicEncrypt } from "crypto";
+import {
+  EncryptOptions,
+  EncryptParameters,
+  EncryptResult,
+  KeyWrapAlgorithm,
+  VerifyOptions,
+  VerifyResult,
+  WrapKeyOptions,
+  WrapResult
+} from "..";
+import { isNode } from "@azure/core-http";
+import { convertJWKtoPEM } from "./conversions";
+import { LocalCryptographyUnsupportedError, LocalSupportedAlgorithmName } from "./models";
+import { SignatureAlgorithm } from "../cryptographyClientModels";
+import { createHash } from "./hash";
+
+/**
+ * An interface representing a local-only cryptography provider.
+ * @internal
+ */
+export interface LocalCryptographyProvider {
+  /**
+   * Encrypts the given plaintext with the specified cryptography algorithm using the provided key.
+   * @param key - The key to use for encryption.
+   * @param encryptParameters - The encryption parameters.
+   * @param options - Additional options.
+   * @internal
+   */
+  encrypt(
+    key: JsonWebKey,
+    encryptParameters: EncryptParameters,
+    options: EncryptOptions
+  ): Promise<EncryptResult>;
+
+  /**
+   * Returns whether this provider supports the given encryption algorithm.
+   * @param algorithm - The algorithm to check.
+   * @internal
+   */
+  isApplicable(algorithm: LocalSupportedAlgorithmName): boolean;
+
+  /**
+   * Wraps the given key using the specified cryptography algorithm and provided key.
+   * @param key - The key to use for wrapping.
+   * @param algorithm - The encryption algorithm to use to wrap the given key.
+   * @param keyToWrap - The key to wrap.
+   * @param options - Additional options.
+   * @internal
+   */
+  wrapKey(
+    key: JsonWebKey,
+    algorithm: KeyWrapAlgorithm,
+    keyToWrap: Uint8Array,
+    options: WrapKeyOptions
+  ): Promise<WrapResult>;
+
+  /**
+   * Verify the signed block of data.
+   * @param key - The key to use for verification.
+   * @param algorithm - The encryption algorithm to use for verification.
+   * @param data - The signed block of data to verify.
+   * @param signature  - The signature to verify the block against.
+   * @param options - Additional options.
+   * @internal
+   */
+  verifyData(
+    key: JsonWebKey,
+    algorithm: SignatureAlgorithm,
+    data: Uint8Array,
+    signature: Uint8Array,
+    options: VerifyOptions
+  ): Promise<VerifyResult>;
+
+  /**
+   * Hashes a block of data using a given algorithm.
+   * @param algorithm - The encryption algorithm to use for hashing.
+   * @param data - The data to hash.
+   * @internal
+   */
+  createHash(algorithm: SignatureAlgorithm, data: Uint8Array): Promise<Buffer>;
+}
+
+/**
+ * An RSA cryptography provider supporting RSA algorithms.
+ */
+export class RsaCryptographyProvider implements LocalCryptographyProvider {
+  ensureValid(operationName: string, key?: JsonWebKey): void {
+    if (!isNode) {
```

Comment:
since this file (and presumably the things it imports) now gets included in browser bundles, how does that affect browser bundle size?

Reply:
Hi Jeff! I will get you an answer but I caught up with Daniel a little while ago and I want to take this refactor a little further so I'll be closing this PR and reopening it when I have all the changes up. I'll also check on the bundle size and if we can keep it from bloating!